homeassistant/components/moon/calendar.py

```diff
@@ -0,0 +1,148 @@
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Moon phase Calendar config entry."""
+
+    obj_moon_phases = get_moon_phases(
+        dt_util.now().date(), (dt_util.now().date() + timedelta(days=30))
+    )
+
+    async_add_entities(
+        [
+            MoonCalendarEntity(
+                config_entry.title,
+                obj_moon_phases,
+                config_entry.entry_id,
+            )
+        ],
+        True,
+    )
+
+
+class MoonCalendarEntity(CalendarEntity):
+    """Representation of a Moon Phase Calendar element."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_translation_key = "moon_phases"
+
+    def __init__(
+        self,
+        name: str,
+        obj_moon_phases: list[dict[str, Any]],
+        unique_id: str,
+    ) -> None:
+        """Initialize MoonCalendarEntity."""
+        self._attr_unique_id = unique_id
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, unique_id)},
+            entry_type=DeviceEntryType.SERVICE,
+            name=name,
+        )
+        self._obj_moon_phases = obj_moon_phases
+
+    @property
+    def event(self) -> CalendarEvent | None:
+        """Return the next upcoming moon    phase."""
+        summary: str
+        for moon_phase in self._obj_moon_phases:
+            if moon_phase["date"] >= dt_util.now().date():
+                summary = moon_phase["phase"]
+                next_moon_phase = (
+                    moon_phase["date"],
+                    summary,
+                    moon_phase["end"],
+                )
+                break
+
+        return CalendarEvent(
+            summary=async_translate_calendar_summary(
+                self.hass,
+                DOMAIN,
+                self._attr_translation_key,
+                next_moon_phase[1],
+            ),
+            start=next_moon_phase[0],
+            end=next_moon_phase[2],
+        )
+
+    async def async_get_events(
+        self, hass: HomeAssistant, start_date: datetime, end_date: datetime
+    ) -> list[CalendarEvent]:
+        """Get all events in a specific time frame."""
+        obj_moon_phases = get_moon_phases(start_date.date(), end_date.date())
+
+        event_list: list[CalendarEvent] = []
+
+        for moon_phase in obj_moon_phases:
+            if moon_phase is not None:
+                event = CalendarEvent(
+                    summary=async_translate_calendar_summary(
+                        self.hass,
+                        DOMAIN,
+                        self._attr_translation_key,
+                        moon_phase["phase"],
+                    ),
+                    start=moon_phase["date"],
+                    end=moon_phase["end"],
+                )
+                event_list.append(event)
+
+        return event_list
+
+
+def async_translate_calendar_summary(
```

Comment:
This doesn't seem like a blessed way to handle translations. (I am not familiar with all translation best practices, but this doesn't seem like its following them)